core/tests/webdriverio_utils/StoryEditorPage.js

```diff
@@ -0,0 +1,182 @@
+  var nodeOutlineFinalizeCheckbox = $('.e2e-test-finalize-outline');
+  var nodeOutlineSaveButton = $('.e2e-test-node-outline-save-button');
+  var createChapterThumbnailButton = $(
+    '.e2e-test-chapter-input-thumbnail .e2e-test-photo-button');
+
+  this.get = async function(storyId) {
+    await browser.url(EDITOR_URL_PREFIX + storyId);
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.returnToTopic = async function() {
+    await general.scrollToTop();
+    await action.click('Return to topic button', returnToTopicButton);
+    await waitFor.pageToFullyLoad();
+  };
+
+  this.updateMetaTagContent = async function(newMetaTagContent) {
+    await action.setValue(
+      'Update Meta Tag Content', storyMetaTagContentField, newMetaTagContent);
+    await action.click('Meta Tag Content label', storyMetaTagContentLabel);
+  };
+
+  this.saveStory = async function(commitMessage) {
+    await action.click('Save Story Button', saveStoryButton);
+    await waitFor.visibilityOf(
+      commitMessageField, 'Commit message modal takes too long to appear.');
+    await commitMessageField.setValue(commitMessage);
+
+    await action.click('Close Save Modal button', closeSaveModalButton);
+    await waitFor.invisibilityOf(
+      closeSaveModalButton,
+      'Commit message modal takes too long to disappear.');
+    await waitFor.pageToFullyLoad();
+    // Wait for the "Save Draft" button to be reset.
+    await waitFor.visibilityOf(
+      saveStoryButton, 'Save Story Button taking too long to appear.');
+    await waitFor.textToBePresentInElement(
+      saveStoryButton, 'Save Draft', 'Story could not be saved.');
+  };
+
+  this.publishStory = async function() {
+    await action.click('Publish Story Button', publishStoryButton);
+  };
+
+  this.unpublishStory = async function() {
+    await action.click('Unpublish Story Button', unpublishStoryButton);
+    await action.click('Close Save Modal button', closeSaveModalButton);
+    await waitFor.invisibilityOf(
+      closeSaveModalButton,
+      'Unpublish message modal takes too long to disappear.');
+  };
+
+  this.createNewChapter = async function(title, explorationId, imgPath) {
+    await general.scrollToTop();
+    await action.click(
+      'Create chapter button takes too long to be clickable.',
+      createChapterButton);
+    await action.setValue(
+      'New chapter title field', newChapterTitleField, title);
+    await action.setValue(
+      'New chapter exploration ID', newChapterExplorationField, explorationId);
+    await workflow.submitImage(
+      createChapterThumbnailButton, thumbnailContainer, imgPath, false);
+    await action.click(
+      'Confirm chapter creation button', confirmChapterCreationButton);
+    await general.scrollToTop();
+  };
+
+  this.navigateToChapterWithName = async function(chapterName) {
+    var chapterNameSelector = $(`span=${chapterName}`);
+    await waitFor.visibilityOf(
+      chapterNameSelector, 'Chapter name is taking too lon to appear');
```

Comment:
Why are you not using chapterTitles.first()?
```suggestion
      chapterNameSelector, 'Chapter name is taking too long to appear');
```

Reply:
After creating the first chapter there is only one element present with selector `$(`span=${chapterName}`)` and as I already said we cannot use all selector for single element, that why as a work around I am just checking whether the needed chapter is present or not.

Reply:
I don't think so we use `span=${chapterName}` as a classname. We have something like `.protractor-test-*`. Also, one thing if we can't use all selector with one element, then how it's working at line 124 in the same file. Are you telling that we can't use `.first()` with all selector if we are having one element.

Reply:
Reverted now using e2e-test classname only as we discussed.